extract.pro

```diff
@@ -160,18 +163,25 @@ for ifib=0,mfib-1 do begin
 ; and shiftme is set
 ; In what follows, dymed and dymed0 are vectors of length nord
 ;  dymed0xyz is dymed0 rebinned to (nx,nord,mfib)
+
   if(shiftme eq 1 and ifib eq 0) then begin
     dymed=dymedian(mom0,mom1,cct)
     if(abs(median(dymed)) gt 0.1) then begin
       logo_nres2,rutname,'INFO','median(dymed)=  '+string(median(dymed),$
-         format='(f6.3)')+'  Shifting...'
+         format='(f6.3)')+'  Shifting...' +'Current iteration:  '+string(current_iter,$
+         format='(f6.0)')
 ;     print,'median(dymed)=',median(dymed),'    Shifting...'
       dymed0=dymed0+dymed
       dymed0xyz=rebin(reform(dymed0,1,nord,1),nx,nord,mfib)
 ;     uu=mom1(1950:2150,*)
 ;     uum=median(uu,dimen=1)
 ;     plot,uum
-      goto,traceloop
+      if current_iter le max_iter then begin
+        current_iter = current_iter+1
+        goto,traceloop
+      endif else begin
+        message, 'Maximum iteration reached !!!',/CONTINUE
```

Comment:
the change in behaviour is now that the previously infinite loop would stall the entire pipeline, but at least bad data would be rejected from the incoming data stream. 

With the CONTINUE now in place, non-converging data would be accepted into the data stream. This seems undesireable as we are letting in known bad data. 

Proposed solution: If iteration does no converge, reject the processed spectrum from further processing - mark in database as bad if that is feasible.